sql/src/main/java/org/apache/druid/sql/avatica/DruidStatement.java

```diff
@@ -142,6 +150,11 @@ public DruidStatement(
     return columns;
   }
 
+  public void setParameters(List<TypedValue> parameters)
+  {
+    this.parameters = parameters;
```

Comment:
It seems that this needs to lock `lock`.. Maybe `parameters` can be passed to `execute()` directly without having this method.

Reply:
good catch, reworked to make `execute` take a list of parameters to avoid messing with the lock